pyomo/opt/tests/base/test_factory.py

```diff
@@ -26,23 +26,23 @@
 old_tempdir = TempfileManager.tempdir
 
 
-class TestWriter(pyomo.opt.AbstractProblemWriter):
+class DummyWriter(pyomo.opt.AbstractProblemWriter):
```

Comment:
Just a suggestion: I think it's more common to refer to these as "mock" objects in Python. I'm fine with `DummyWriter` but we may want to consider `MockWriter`